src/TwitchLib.Communication/TwitchLib.Communication.csproj

```diff
@@ -1,24 +1,30 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
-
-  <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
-    <VersionPrefix>1.0.6</VersionPrefix>
-    <VersionSuffix>$(VersionSuffix)</VersionSuffix>
-    <Authors>swiftyspiffy, Prom3theu5, Syzuna, LuckyNoS7evin</Authors>
-    <Company>swiftyspiffy, Prom3theu5, Syzuna, LuckyNoS7evin</Company>
-    <Description>Connection library used throughout TwitchLib to replace third party depedencies.</Description>
-    <Copyright>Copyright 2022</Copyright>
-    <PackageLicenseUrl>https://opensource.org/licenses/MIT</PackageLicenseUrl>
-    <PackageProjectUrl>https://github.com/TwitchLib/TwitchLib.Communication</PackageProjectUrl>
-    <PackageIconUrl>https://cdn.syzuna-programs.de/images/twitchlib.png</PackageIconUrl>
-    <RepositoryUrl>https://github.com/TwitchLib/TwitchLib.Communication</RepositoryUrl>
-    <RepositoryType>Git</RepositoryType>
-    <PackageTags>twitch twitchlib library irc chat c# csharp api events pubsub net standard 2.0</PackageTags>
-    <PackageReleaseNotes>Fix reconnect loop on disconnect</PackageReleaseNotes>
-    <NeutralLanguage>en-US</NeutralLanguage>
-    <AssemblyVersion>1.0.6</AssemblyVersion>
-    <FileVersion>1.0.6</FileVersion>
-    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
-  </PropertyGroup>
-
-</Project>
+    <PropertyGroup>
+        <TargetFramework>netstandard2.0</TargetFramework>
+        <VersionPrefix>2.0.0</VersionPrefix>
+        <LangVersion>8.0</LangVersion>
+        <Nullable>enable</Nullable>
```

Comment:
Are you sure you want to enable nullable reference types? I think you just added whole lot of extra complexity to your PR. You must append `?` to all reference types that are nullable. As result of this small change, there are a lot of warning across your code (for example assigning null to `CancellationTokenSource`)